packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,456 @@
+          ...defaultOptions,
+          recommendationsPerRow: 'invalid',
+        });
+        await flushPromises();
+
+        const initializationError = element.shadowRoot.querySelector(
+          selectors.initializationError
+        );
+
+        expect(initializationError).not.toBeNull();
+      });
+
+      it('should diplay an error message when recommendationsPerRow is a negative number', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          recommendationsPerRow: -1,
+        });
+        await flushPromises();
+
+        const initializationError = element.shadowRoot.querySelector(
+          selectors.initializationError
+        );
+
+        expect(initializationError).not.toBeNull();
+      });
+    });
+  });
+
+  describe('when an initialization error occurs', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    it('should display the error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('when the component is initialized successfully', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+    });
+
+    describe('when there are no recommendations', () => {
+      it('should not display a recommendation list', async () => {
+        recommendationListState = {
+          isLoading: false,
+          recommendations: [],
+          error: null,
+          searchResponseId: null,
+        };
+        const element = createTestComponent();
+        await flushPromises();
+
+        const recommendationContainer = element.shadowRoot.querySelector(
+          selectors.gridContainer
+        );
+
+        expect(recommendationContainer).toBeNull();
+        const recommendationListItems = element.shadowRoot.querySelectorAll(
+          selectors.recommendationListItem
+        );
+        expect(recommendationListItems.length).toBe(0);
+      });
+    });
+
+    describe('when there are recommendations', () => {
+      describe('when the variant value is #grid', () => {
+        it('should properly display the recommendations with default options', async () => {
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant: 'grid',
+          });
+          await flushPromises();
+
+          const recommendationContainer = element.shadowRoot.querySelector(
+            selectors.gridContainer
+          );
+          const recommendationItems = element.shadowRoot.querySelectorAll(
+            selectors.recommendationListItem
+          );
+
+          expect(recommendationContainer).not.toBeNull();
+          expect(recommendationItems.length).toBe(recommendationsArray.length);
```

Comment:
why are we not looping over `recommendationItems ` to confirm that each recommendation has been correctly passed to a quantic result component?